tests/components/utility_meter/test_sensor.py

```diff
@@ -231,6 +231,133 @@ async def test_state(hass: HomeAssistant, yaml_config, config_entry_config) -> N
     assert state is not None
     assert state.state == "unavailable"
 
+    if not yaml_config:
+        result = await hass.config_entries.options.async_init(config_entry.entry_id)
+        result = await hass.config_entries.options.async_configure(
+            result["flow_id"],
+            user_input={
+                "periodically_resetting": True,
+                "always_available": True,
+                "source": entity_id,
+            },
+        )
+
+        await hass.async_block_till_done()
+
+        # test unavailable state
+        state = hass.states.get("sensor.energy_bill_offpeak")
+        assert state is not None
+        assert state.state == "3"
+
+        # test unknown state
+        hass.states.async_set(
+            entity_id, None, {ATTR_UNIT_OF_MEASUREMENT: UnitOfEnergy.KILO_WATT_HOUR}
+        )
+        await hass.async_block_till_done()
+        state = hass.states.get("sensor.energy_bill_offpeak")
+        assert state is not None
+        assert state.state == "3"
+
+
+@pytest.mark.parametrize(
+    ("yaml_config", "config_entry_config"),
+    (
+        (
+            {
+                "utility_meter": {
+                    "energy_bill": {
+                        "source": "sensor.energy",
+                        "always_available": True,
+                    }
+                }
+            },
+            None,
+        ),
+        (
+            None,
+            {
+                "cycle": "none",
+                "delta_values": False,
+                "name": "Energy bill",
+                "net_consumption": False,
+                "offset": 0,
+                "periodically_resetting": True,
+                "source": "sensor.energy",
+                "tariffs": [],
+                "always_available": True,
+            },
+        ),
+    ),
+)
+async def test_state_always_available(
```

Comment:
Same here. Please create a test for yaml and one for config flow. You can create a protected function for the shared code, so you don't need to duplicate it

Reply:
The pattern used in testing this integration is to test the sensor coming from the yaml configuration or configuration entry in the same test (see `test_state`), I just kept this pattern.

Do you want me to separate the `test_state_always_available` test into two to handle the yaml configuration and configuration input in separate tests? I would like you to confirm this before I change it.

Reply:
Okay we can leave it for now and we should refactor the whole file in the future.